### PR TITLE
Pokenav decomp again

### PR DESCRIPTION
**Pull request by Phlosioneer**
My progress so far on pokenav. I need to decompile ~10 more functions, so that I can figure out some of the confusing struct nesting.

### CODE REVIEW

**Comment by Phlosioneer** [comment]
Pokenav decomp is finally ready for review! I was able to figure out the file breaks for pokenav.s, and started documenting a lot of the decompiled code.

match_call_ui could use some more work, but I'd rather start the process of reviewing first.

File: src/pokenav_match_call_ui.c [hunk 1]
```diff
@@ -0,0 +1,1266 @@
...
+{
+    1, 4, 5
+};
+
+static const u8 sUnknown_0861FBF7[] =
+{
+    2, 4, 6, 7, 0
+};
+
+static const struct CompressedSpriteSheet sMatchcallArrowSpriteSheet[] =
```
Comment (huderlem) [inline src/pokenav_match_call_ui.c]: This name should be plural

File: src/pokenav_match_call_ui.c [hunk 2]
```diff
@@ -0,0 +1,1266 @@
...
+{
+    {
+        .data = sMatchcallArrowSpriteSheetData,
+        .size = 192,
+        .tag = 0xA
+    }
+};
+
+static const struct SpritePalette sMatchcallArrowPalette[] =
```
Comment (huderlem) [inline src/pokenav_match_call_ui.c]: plural

File: src/pokenav_match_call_ui.c [hunk 3]
```diff
@@ -0,0 +1,1266 @@
...
+            return 4;
+        default:
+            return 4;
+    }
+}
+
+void sub_81C82E4(struct UnknownSubStruct_81C81D4 *a0)
+{
+    u16 v1 = (a0->unk0.unk0.unk1 << 12) | a0->unk0.unk0.unk6;
+    // TODO: When #553 is merged, use a PALETTE_NUM_TO_FILL_VALUE(1) macro here...
```
Comment (huderlem) [inline src/pokenav_match_call_ui.c]: can do this TODO now

File: sym_ewram.txt [hunk 4]
```diff
@@ -168,17 +168,8 @@ gUnknown_02022D00: @ 2022D00
...
-gUnknown_0203CF3C: @ 203CF3C
-	.space 0x4
-
-gUnknown_0203CF40: @ 203CF40
-	.space 0x4
-
-gUnknown_0203CF44: @ 203CF44
-	.space 0x4
-
+    .include "src/pokenav.o"
```
Comment (huderlem) [inline sym_ewram.txt]: tabs

**Comment by Sewef** [comment]
pokenav_match_call_data.c
```
static const struct MatchCallStruct4 sMayMatchCallHeader =
{
    .type = 4,
    .gender = MALE,
    .flag = FLAG_ENABLE_RIVAL_MATCH_CALL,
    .desc = gMayBrendanMatchCallDesc,
    .name = gExpandedPlaceholder_May,
    .textData = sMayTextScripts
};
```

```
static const struct MatchCallStruct4 sBrendanMatchCallHeader =
{
    .type = 4,
    .gender = FEMALE,
    .flag = FLAG_ENABLE_RIVAL_MATCH_CALL,
    .desc = gMayBrendanMatchCallDesc,
    .name = gExpandedPlaceholder_Brendan,
    .textData = sBrendanTextScripts
};
```

Misslabelling : Brendan is now a girl and May is a boy.

**Comment by Phlosioneer** [comment]
This PR is updated.

**Pull request merged**